libraries/hotstuff/qc_chain.cpp

```diff
@@ -497,6 +491,8 @@ namespace eosio { namespace hotstuff {
       //check for leader change
       leader_rotation_check();
 
+      ++_state_version;
```

Comment:
This is not reliable. For example the call above to ` update(proposal);` contains  EOS_ASSERT macros, which could throw exceptions causing `++_state_version;` not to be executed (even though we modified the state in `insert_proposal` above.
I'd recomment using `auto increment_version = fc::make_scoped_exit([this]() { ++_state_version; };` where the state is modified **first**.